api-gateway/src/main/resources/static/scripts/fragments/nav.html

```diff
@@ -1,10 +1,25 @@
 <script>
-    function purgeUser() {
-        localStorage.removeItem("username")
-        localStorage.removeItem("email")
-        localStorage.removeItem("UUID")
-        localStorage.removeItem("roles")
-}
+    function logoutUser() {
+        logoutUser.$inject = ['$http'];
+        var $http = angular.injector(['ng']).get('$http');
+        $http({
+            method: 'POST',
+            url: '/api/gateway/users/logout'
+        }).then(function successCallback(response) {
+            alert("Logout successful")
+            localStorage.removeItem("username")
+            localStorage.removeItem("email")
+            localStorage.removeItem("UUID")
+            localStorage.removeItem("roles")
+            window.location.href = '/#!/login';
+        }, function errorCallback(response) {
+            alert("Logout Error: Session may have expired. Redirecting to login page")
+            localStorage.removeItem("username")
+            localStorage.removeItem("email")
+            localStorage.removeItem("UUID")
+            localStorage.removeItem("roles")
+            window.location.href = '/#!/login';
+        });    }
 </script>
```

Comment:
Nice work handling the expired sessions, did you test it to make sure it works ?

Reply:
Yes, I did do one test for an expired session in the ApiGatewayControllerTest:
![SessionExpiredTest](https://github.com/cgerard321/champlain_petclinic/assets/77691550/8eb842ba-2d5c-4727-85e8-534fcd2d4218)